internal/launcher/launcher.go

```diff
@@ -20,6 +20,20 @@ func Execute() {
 	gs.OSExit(newLauncher(gs).launch())
 }
 
+func ExecuteForTesting(gs *state.GlobalState) {
```

Comment:
I would pass a `testintg.T` argument and make the functions fail instead of panic because panic output tends to be very verbose.

Reply:
I agree. However, I don't plan to keep it in this form, I mostly did it for simplicity. As soon as we have a final version for #4671 then I will move to a real mock directly to some test package. In the end, it is what the component is doing.

But before, I wanted to finish the review of other parts.